ddtrace/llmobs/_llmobs.py

```diff
@@ -147,25 +158,26 @@ def enable(
         config._llmobs_ml_app = ml_app or config._llmobs_ml_app
 
         # validate required values for LLMObs
-        if not config._dd_api_key:
-            raise ValueError(
-                "DD_API_KEY is required for sending LLMObs data. "
-                "Ensure this configuration is set before running your application."
-            )
-        if not config._dd_site:
-            raise ValueError(
-                "DD_SITE is required for sending LLMObs data. "
-                "Ensure this configuration is set before running your application."
-            )
         if not config._llmobs_ml_app:
             raise ValueError(
                 "DD_LLMOBS_ML_APP is required for sending LLMObs data. "
                 "Ensure this configuration is set before running your application."
             )
 
         if agentless_enabled or asbool(os.getenv("DD_LLMOBS_AGENTLESS_ENABLED", "false")):
-            os.environ["DD_LLMOBS_AGENTLESS_ENABLED"] = "1"
+            # validate required values for agentless LLMObs
+            if not config._dd_api_key:
+                raise ValueError(
+                    "DD_API_KEY is required for sending LLMObs data. "
+                    "Ensure this configuration is set before running your application."
+                )
+            if not config._dd_site:
+                raise ValueError(
+                    "DD_SITE is required for sending LLMObs data. "
+                    "Ensure this configuration is set before running your application."
+                )
 
+            os.environ["DD_LLMOBS_AGENTLESS_ENABLED"] = "1"
```

Comment:
Is there a codepath that depends on this environment variable being toggled to true somewhere?
I think `__init__` is called before [here](https://github.com/DataDog/dd-trace-py/blob/5c4fd668b2dd25fff6a6d2ad97a8394174fc7aee/ddtrace/llmobs/_llmobs.py#L791C1-L791C28) before `enable` is called, which is the only place i see this env var being read

Reply:
I think once https://github.com/DataDog/dd-trace-py/pull/9912 we can use `config._llmobs_agentless_enabled` as the source of truth, either in a follow-up PR or in this PR

Reply:
I think it's used in the re-init [here](https://github.com/DataDog/dd-trace-py/blob/5c4fd668b2dd25fff6a6d2ad97a8394174fc7aee/ddtrace/llmobs/_llmobs.py#L194).
If the env var is not set and we explicitly pass in `agentless_enabled` we set the env var and re-init the instance.
And I didn't touch the code - just moved it from line 167 to here.

Reply:
that makes sense, thanks!